roxie/ccd/ccdserver.cpp

```diff
@@ -24656,9 +24656,9 @@ class CRoxieServerKeyedJoinBase : public CRoxieServerActivity, implements IRecor
 
     void processCompletedGroups()
     {
+        CriticalBlock c(groupsCrit);
```

Comment:
previously, the loop and position of the crit, meant it would effectively yield, per iteration around the loop.
Are there any concerns that it now holds the crit until all pending groups are processed?